kv/util/engine_util/engines.go

```diff
@@ -38,11 +38,14 @@ func (en *Engines) WriteRaft(wb *WriteBatch) error {
 }
 
 func (en *Engines) Close() error {
-	if err := en.Kv.Close(); err != nil {
-		return err
-	}
-	if err := en.Raft.Close(); err != nil {
-		return err
+	dbs := []*badger.DB{en.Kv, en.Raft}
+	for _, db := range dbs {
+		if db == nil {
```

Comment:
What's the case that the db is nil

Reply:
> What's the case that the db is nil

Engines can simulate column families .
User can use it to easily implement storage (stand-along or distribution) with column families.
When Implementing stand-along storage by Engines, Engines.Raft is nil .
From the perspective of Defensive Programming, it is necessary to judge whether db is nil .